### PR TITLE
using Feature.COLOR_HALF_FLOAT

### PR DESCRIPTION
**Pull request by troublemaker52025**
Re: cocos-creator/3d-tasks#

Changelog:
 *

### CODE REVIEW

**Comment by troublemaker52025** [comment]
https://github.com/cocos-creator/3d-tasks/issues/6914

File: cocos/core/pipeline/pipeline-ubo.ts [hunk 1]
```diff
@@ -183,7 +183,7 @@ export class PipelineUBO {
                 Mat4.multiply(matShadowViewProj, matShadowViewProj, matShadowView);
                 Mat4.toArray(sv, matShadowViewProj, UBOShadow.MAT_LIGHT_VIEW_PROJ_OFFSET);
 
-                const isTextureHalfFloat = device.hasFeature(Feature.TEXTURE_HALF_FLOAT);
+                const isTextureHalfFloat = device.hasFeature(Feature.COLOR_HALF_FLOAT);
                 const linear = (shadowInfo.linear && isTextureHalfFloat) ? 1.0 : 0.0;
```
Comment (caryliu1999) [inline cocos/core/pipeline/pipeline-ubo.ts]: 封装个统一的接口吧，别每次改好几个地方

Comment (troublemaker52025) [reply cocos/core/pipeline/pipeline-ubo.ts]: sure

**Comment by troublemaker52025** [comment]
https://github.com/cocos-creator/engine-native/pull/3496

**Review by caryliu1999 — APPROVED** [review]

File: cocos/core/gfx/base/device.ts [hunk 2]
```diff
@@ -412,4 +412,13 @@ export abstract class Device {
...
         return this._features[feature];
     }
+
+    /**
+     * @en Whether to support half-float precision texture format.
+     * @zn 是否支持 half-float 精度的纹理格式。
+     * @param feature the support precision texture format.
+     */
+    public isSupportHalfFloat () : boolean {
+        return this._features[Feature.COLOR_HALF_FLOAT] && this._features[Feature.TEXTURE_HALF_FLOAT];
```
Comment (YunHsiao) [inline cocos/core/gfx/base/device.ts:422]: 🤦

**Pull request merged**